cpp/src/io/orc/reader_impl.cu

```diff
@@ -1262,11 +1262,17 @@ table_with_metadata reader::impl::read(size_type skip_rows,
 
   out_metadata.schema_info = std::move(schema_info);
 
-  for (const auto& meta : _metadata.per_file_metadata) {
-    for (const auto& kv : meta.ff.metadata) {
-      out_metadata.user_data.insert({kv.name, kv.value});
-    }
-  }
+  std::transform(_metadata.per_file_metadata.cbegin(),
+                 _metadata.per_file_metadata.cend(),
+                 std::back_inserter(out_metadata.per_file_user_data),
+                 [](auto& meta) {
+                   std::map<std::string, std::string> kv_map;
+                   for (auto& kv : meta.ff.metadata) {
+                     kv_map[kv.name] = kv.value;
+                   }
```

Comment:
Here orc/reader_impl.cu range based for loop is used, in parquet/reader_impl.cu, `std::transform` is used. 
For uniformity, sticking to one style would be nice.